modules/apps/object/object-service/src/main/java/com/liferay/object/internal/field/business/type/RelationshipObjectFieldBusinessType.java

```diff
@@ -77,6 +95,67 @@ public Set<String> getRequiredObjectFieldSettingsNames() {
 				NAME_OBJECT_RELATIONSHIP_ERC_FIELD_NAME);
 	}
 
+	@Override
+	public Object getValue(ObjectField objectField, Map<String, ?> values)
+		throws PortalException {
+
+		Object value = values.get(objectField.getName());
+
+		if (!Objects.equals(
+				objectField.getRelationshipType(),
+				ObjectRelationshipConstants.TYPE_ONE_TO_MANY) ||
+			(GetterUtil.getLong(value) > 0)) {
+
+			return value;
+		}
+
+		String externalReferenceCode = GetterUtil.getString(
+			values.get(
+				ObjectFieldSettingValueUtil.getObjectFieldSettingValue(
+					objectField,
+					ObjectFieldSettingConstants.
+						NAME_OBJECT_RELATIONSHIP_ERC_FIELD_NAME)));
+
+		if (Validator.isNull(externalReferenceCode)) {
+			return null;
+		}
+
+		ObjectRelationship objectRelationship =
+			_objectRelationshipPersistence.findByObjectFieldId2(
+				objectField.getObjectFieldId());
+
+		ObjectDefinition objectDefinition =
+			_objectDefinitionPersistence.findByPrimaryKey(
+				objectRelationship.getObjectDefinitionId1());
+
+		if (objectDefinition.isSystem()) {
+			SystemObjectDefinitionMetadata systemObjectDefinitionMetadata =
+				_systemObjectDefinitionMetadataTracker.
+					getSystemObjectDefinitionMetadata(
+						objectDefinition.getName());
+
+			BaseModel<?> baseModel =
+				systemObjectDefinitionMetadata.
+					getBaseModelByExternalReferenceCode(
+						externalReferenceCode, objectDefinition.getCompanyId());
```

Comment:
We are treating the exception when it is a custom object definition. However, I wonder if we should treat this situation as well?  What do you think? :thinking:

Reply:
In each systemObjectDefinitionMetadata, we call the "finder" method, which can throw an exception.

Reply:
I'm treating the exception when it is a custom object because I need to create the `ObjectEntry` if it doesn't exist.

I don't now how we could treat this situation for System Objects, Any suggestion?